### ISSUE TITLE
[Bug]: How to analyse issue#5829

### ISSUE THREAD

**Issue by GaoYizhuo2000** [amyWenhui/apps-android-commons#5]
### Summary

On the pending upload screen, if the user pauses any ongoing image upload and then discards it from the pending uploads, then that image is considered queued even if the user uploads the image again.

### Steps to reproduce

1.Upload an image normally from the custom selector.
2.After the uploads start, the app navigates to the contributions screen
3.Navigate to the pending uploads screen and pause the upload you just started.
4.After pausing, click the cross icon to cancel the upload.
5.Again upload the same image and navigate to the pending uploads screen.

### Expected behaviour

The upload should fresh start again.

### Actual behaviour

The upload is queued.

### Device name

Samsung A14

### Android version

Android 14

### Commons app version

5.0.2

### Would you like to work on the issue?

None

**Comment by GaoYizhuo2000** [comment]
Guys, i think to find a out the reason of this bug, we need to do a thorough reproduction, since my first feeling about this bug is that there might be just some internet delay or something related to threads handling that result in this bug. So maybe if the user give enough time for this app to handle the pause and deletion, the future same upload will start as expected.

**Comment by GaoYizhuo2000** [comment]
I just tried to reproduce this bug and found that it doesn't matter if the user upload the same image again or not, the fact is that once the pause button is clicked, all the upload will be paused, including future upload, and until the restart button is clicked will the uploads start again.

**Comment by GaoYizhuo2000** [comment]
So i for now i think it's better to point this out under the master repo and see what the maintainer will say about this. And at the same time I will try to find the source code that relates to this.

**Comment by Bei-Jin-lab** [comment]
Yes, I also find that when I run this app. Maintainer has left a comment, also agreed your statement, so we can fix the bug about pause button

**Comment by Bei-Jin-lab** [comment]
The original issue comment https://github.com/commons-app/apps-android-commons/issues/5829

**Comment by GaoYizhuo2000** [comment]
yes, i have already fixed this bug and make the future upload start automatically when after pausing and deleting all the current uploads. And we can see it as an agreement on my redefining of this bug although after reading my comment the maintainer said the bug itself  no longer exists. And I hope that my pr will be merged anyway.

**Comment by FanKiona** [comment]
Maybe we can emphasize that although this change is no longer a direct fix to the original issue, it serves as a logical optimization that improves the overall code quality and user experience. This might increase the likelihood of the maintainer accepting it as an optimization merge.
